tonilopezmr/day4/PassphraseValidator.kt

```diff
@@ -0,0 +1,73 @@
+package day4
+
+import org.junit.Assert.*
+import org.junit.Test
+import java.io.File
+import java.util.*
+
+class PassphraseValidator {
+
+  fun validI(passphrase: String): Boolean {
+    val words = passphrase
+        .split(" ")
+    return words
+        .foldIndexed(true) { idx, acc, a ->
+          if (words.subList(idx + 1, words.size).contains(a)) {
+            false
+          } else {
+            acc
+          }
+        }
+  }
```

Comment:
I guess you could simplify this to

```kotlin
return words.all { word -> words.count { it == word } <= 1 }  
```

Reply:
wooooot?

Reply:
`all` checks all the elements in a collection fulfill a predicate.

`count` counts how many items in a collection fulfill a predicate.

Reply:
too much magic behind this methods

Reply:
Declarative magic 😉